src/Backend/Modules/Locale/Engine/LocaleAnalyser.php

```diff
@@ -25,11 +25,47 @@ public function __construct(string $application, string $modulesPath, array $ins
 
     public function findMissingLocale(string $language): array
     {
+        $this->missingLocale = [];
         $moduleFiles = $this->findModuleFiles();
+        $existingLocale = $this->getExistingLocaleForLanguage($language);
+
         $locale = $this->findLocaleInFiles($moduleFiles);
 
-        // @TODO
-        return [];
+        $nonExisting = [];
+        foreach ($locale as $moduleName => $module) {
+            foreach ($module as $filename => $file) {
+                $file['locale'] = $this->recursiveArrayDiff($file['locale'], $existingLocale);
+
+                foreach ($file['locale'] as $type => $modules) {
+                    foreach ($modules as $key => $translations) {
+                        foreach ($translations as $translationName) {
+                            $nonExistingKey = $this->application . $translationName . $type . $key;
+
+                            if (array_key_exists($nonExistingKey, $nonExisting)) {
+                                $usedInFiles = unserialize($nonExisting[$nonExistingKey]['used_in']);
+                                $usedInFiles[] = $file['file'];
+                                $nonExisting[$nonExistingKey]['used_in'] = serialize($usedInFiles);
+
+                                continue;
+                            }
+
+                            $nonExisting[$nonExistingKey] = [
```

Comment:
Would it be an idea to create a value object (of some sort) to hold this data? Seems a bit more robust than a simple array. Not saying you have to do it, I just think it may be a good idea

Reply:
As discussed in person: I think it is overkill and not very useful since the array is also the source of a datagrid